src/lib/Libutil/misc_utils.c

```diff
@@ -1035,23 +1035,29 @@ validate_ext_auth_data(int auth_type, void *data, int data_len, char *ebuf, int
 
 /**
  *
- *	@brief break apart a comma delimited string into an array of strings
+ *	@brief break apart a delimited string into an array of strings
  *
- *	  @param[in] strlist - the comma delimited string
+ *	@param[in] strlist - the delimited string
+ *	@param[in] sep - the separator character
  *
  *	@return char **
  *
+ *	@note
+ *		The returned array of strings has to be freed by the caller.
  */
 char **
-break_comma_list(char *strlist)
+break_delimited_str(char *strlist, char delim)
 {
+	char sep[2] = "\0";
```

Comment:
If the intent is to create a NULL terminated string with delim being the only character, you want to initialize this as...
char sep[2] = { '\0', '\0'};

Otherwise, you could sprintf(sep, "%c", delim) below.

What you have now is NOT null terminated.

Reply:
It is null terminated, because the compiler will add the terminating null at the end, since it's a string of length 1.

It does look a little weird though.

Reply:
It's more than weird, it's wrong.

While the compiler may terminate the string literal for us, the string itself doesn’t contain what it should. If you pass comma as the delimiter then sep has the value “,0” which gets passed to strtok_r().

Also, we’re not flagging an error when strdup() fails. We end up returning NULL and pbs_errno is unchanged. Should we not be initializing pbs_errno to PBSE_NONE if the caller will be checking it?

IMHO, this code change is incorrect. Further, there are no test logs attached so it should not have been merged. I’d like to request this change be reverted.

Reply:
Mike I still don't see a problem. If a comma is passed to this function then wouldn't a null terminated comma be passed to strtok_r? With byte order being ,\0
Also while strdup failure is something we could check that is not something Nithin added. 
The code has been existing for as long as I remember and I didn't ask for an additional test because it gets exercised in all the cases where scheduler parses resources string in every sched startup

Reply:
Mike I still don't see a problem. If a comma is passed to this function then wouldn't a null terminated comma be passed to strtok_r? With byte order being ,\0
Also while strdup failure is something we could check that is not something Nithin added. 
The code has been existing for as long as I remember and I didn't ask for an additional test because it gets exercised in all the cases where scheduler parses resources string in every sched startup

Reply:
So I may have misunderstood one aspect of what you are saying. I was thinking that "\0" would result in sep[0] being '\' and sep[1] being '0'. That is not the case. Since the string literal implies a null terminated string, are we certain that the '\0' gets assigned to sep[1]?

I would feel much better if the initialization was done explicitly, without assuming the null terminator gets copied.

There is still the issue of initializing pbs_errno and checking the return code strdup(). Whether Nithin added it or not, it should be checked. Technically, this is a new function even though most of it was copied from break_comma_list().

Reply:
`"\0"` is a character array of length 2; the first character is `'\0'`, and the second character is the null terminating byte, which is also `'\0'`.

What we end up with is:
```C
sep[0] = '\0'
sep[1] = '\0'
```